x/poolmanager/taker_fee.go

```diff
@@ -134,7 +134,7 @@ func (k Keeper) chargeTakerFee(ctx sdk.Context, tokenIn sdk.Coin, tokenOutDenom
 
 	// Determine if eligible to bypass taker fee.
 	if osmoutils.Contains(reducedFeeWhitelist, sender.String()) {
-		return tokenIn, sdk.Coin{}, nil
+		return tokenIn, sdk.NewCoin(tokenIn.Denom, osmomath.ZeroInt()), nil
```

Comment:
nit: can be pre-allocated and coin initialized as `sdk.Coin{Denom: tokenIn.Denom, Amount: zero}`

Reply:
For sure, would your suggestion be to have a var at the top of the file or in a `init()` function?

Reply:
Var at the top is enough IMO. This is a minor optimization to avoid reallocating the int since coin operations allocate a new copy under the hood.

![image](https://github.com/user-attachments/assets/0ffa066c-9878-489f-8115-f1bed5ef0810)

Reply:
Ace, ty